src/ClassDiscovery.php

```diff
@@ -2,126 +2,117 @@
 
 namespace Http\Discovery;
 
-use Puli\Discovery\Api\Discovery;
+use Http\Discovery\Exception\DiscoveryFailedException;
+use Http\Discovery\Exception\StrategyUnavailableException;
 
 /**
  * Registry that based find results on class existence.
  *
  * @author David de Boer <david@ddeboer.nl>
  * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
  */
 abstract class ClassDiscovery
 {
     /**
-     * @var GeneratedPuliFactory
+     * A list of strategies to find classes.
+     *
+     * @var array
      */
-    private static $puliFactory;
+    public static $strategies = [
+        Strategy\Puli::class,
+        Strategy\HttpClients::class,
```

Comment:
Actually I wonder if we should make these three (HttpClients, Guzzle, Diactoros) to be one strategy. Since discovery is done based on type, I see no harm in having them in one array. If this is possible, it invalidates my comment a few lines above